### PR TITLE
fix(core): sanitize narrated fetch URLs

### PR DESCRIPTION
**Pull request by chaliy**
### Motivation
- The previous `url_display` used ad-hoc string splitting which treated credential-like `user:pass@` fragments as path text for protocol-relative or malformed URLs, allowing userinfo to be leaked into pre-execution tool narration. 
- `web_fetch` narration emits the displayed URL before execution/validation, so this created a new leakage path for secrets embedded in URL arguments.

### Description
- Replace ad-hoc splitting with real URL parsing using `url::Url` and normalize scheme-relative inputs by prefixing an `https://` base before parsing. 
- When parsing fails or no host is present, return an empty display so narration falls back to the bare verb rather than showing unsafe substrings. 
- Construct the displayed form as `host[:port] + path` while stripping scheme, userinfo, query, and fragment, and truncate with the existing `truncate` helper. 
- Add regression tests that cover credential-bearing absolute, protocol-relative, and empty-authority/malformed URL forms exercised via `url_display` and `narrate_web_fetch`.

### Testing
- Ran `cargo fmt --check` which passed. 
- Ran focused unit tests for the tool narration module with `cargo test -p everruns-core --lib tool_narration::tests::` and iterated once to correct an assertion; final run reported all module tests passing (`9 passed; 0 failed`). 
- Repository remote was unavailable in this environment, so verification and changes were validated against the current local branch HEAD.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a3475a58698832bafa12768fd1dc5f5)

### CODE REVIEW

File: crates/core/src/tool_narration.rs [hunk 1]
```diff
@@ -890,6 +889,14 @@ mod tests {
             "example.com/path"
         );
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/core/src/tool_narration.rs:890]: Added `web_fetch_helper_falls_back_to_bare_verb_for_malformed_url` (commit 07271be), which drives `narrate_web_fetch` with three malformed/secret-bearing URLs (unparseable, scheme-only with no host, and a `user:pass@` form without a scheme) and asserts the narration is exactly the bare verb (`Fetch URL` / `Fetched URL`) and contains no `secret`/`s3cret` substring.

---
_Generated by [Claude Code](https://claude.ai/code)_
